front/components/assistant_builder/AssistantBuilder.tsx

```diff
@@ -99,20 +99,23 @@ const SPIRIT_AVATAR_URLS = SPIRIT_AVATAR_FILES.map((f) =>
 
 // Actions
 
-const ACTION_MODES = [
-  "GENERIC",
-  "RETRIEVAL_SEARCH",
+const BASIC_ACTION_MODES = ["GENERIC", "RETRIEVAL_SEARCH"] as const;
+const ADVANCED_ACTION_MODES = [
   "RETRIEVAL_EXHAUSTIVE",
   "DUST_APP_RUN",
   "DATABASE_QUERY",
 ] as const;
-type ActionMode = (typeof ACTION_MODES)[number];
+
+type ActionMode =
+  | (typeof BASIC_ACTION_MODES)[number]
+  | (typeof ADVANCED_ACTION_MODES)[number];
+
 const ACTION_MODE_TO_LABEL: Record<ActionMode, string> = {
   GENERIC: "No action",
-  RETRIEVAL_SEARCH: "Search Data Sources",
-  RETRIEVAL_EXHAUSTIVE: "Process Data Sources",
-  DUST_APP_RUN: "Run Dust App",
-  DATABASE_QUERY: "Query Database",
+  RETRIEVAL_SEARCH: "Search in data sources",
```

Comment:
Nit: capitalization seems to differ between RETRIEVAL and DUST_APP/DATABASE_QUERY. I'm in favor of capitalized but I know this is also something on which people have strong ideas?

Reply:
Indeed thanks for noticing! 🙇🏻‍♀️ 
I have no strong opinion on this, I put everything in lower case to be consistent with the following help content, where we use "data source" in lower case.